src/test/java/com/powsybl/openloadflow/sa/LfActionTest.java

```diff
@@ -172,12 +189,69 @@ void testHvdcAction2() {
+            assertFalse(lfHvdcAction.apply(lfNetwork, null, acParameters.getNetworkParameters()));
+
+            LfAction lfHvdcAction2 = LfActionUtils.createLfAction(hvdcAction2, network, acParameters.getNetworkParameters().isBreakers(), lfNetwork);
+            assertFalse(lfHvdcAction2.apply(lfNetwork, null, acParameters.getNetworkParameters()));
+        }
+    }
+
+    @Test
+    void testLfAreaInterchangeTargetAction() {
+
+        AreaInterchangeTargetAction targetAction = new AreaInterchangeTargetActionBuilder()
+            .withId("action")
+            .withAreaId("a1")
+            .withTarget(20.0)
+            .build();
+
+        AreaInterchangeTargetAction invalidTargetAction = new AreaInterchangeTargetActionBuilder()
+            .withId("action")
+            .withAreaId("DUMMMY")
+            .withTarget(20.0)
+            .build();
+
+        Network network = MultiAreaNetworkFactory.createTwoAreasWithXNode();
+
+        var matrixFactory = new DenseMatrixFactory();
+
+        // With area interchange target control enabled
+        AcLoadFlowParameters acParameters = OpenLoadFlowParameters.createAcParameters(network,
+            new LoadFlowParameters(), new OpenLoadFlowParameters(), matrixFactory, new NaiveGraphConnectivityFactory<>(LfBus::getNum), true, false);
+        acParameters.getNetworkParameters().setAreaInterchangeControl(true);
+        try (LfNetworkList lfNetworks = Networks.load(network, acParameters.getNetworkParameters(), new LfTopoConfig(), ReportNode.NO_OP)) {
+            LfNetwork lfNetwork = lfNetworks.getLargest().orElseThrow();
+
+            LfAction lfAreaTargetAction = LfActionUtils.createLfAction(targetAction, network, acParameters.getNetworkParameters().isBreakers(), lfNetwork);
+            assertTrue(lfAreaTargetAction.apply(lfNetwork, null, acParameters.getNetworkParameters()));
+
+            LfAction lfAreaTargetAction2 = LfActionUtils.createLfAction(invalidTargetAction, network, acParameters.getNetworkParameters().isBreakers(), lfNetwork);
+            assertFalse(lfAreaTargetAction2.apply(lfNetwork, null, acParameters.getNetworkParameters()));
+        }
+
+        // With area interchange target control disabled
+        acParameters.getNetworkParameters().setAreaInterchangeControl(false);
+        try (LfNetworkList lfNetworks = Networks.load(network, acParameters.getNetworkParameters(), new LfTopoConfig(), ReportNode.NO_OP)) {
+            LfNetwork lfNetwork = lfNetworks.getLargest().orElseThrow();
+            acParameters.getNetworkParameters().setAreaInterchangeControl(false);
+
+            LfAction lfAreaTargetAction = LfActionUtils.createLfAction(targetAction, network, acParameters.getNetworkParameters().isBreakers(), lfNetwork);
+            assertFalse(lfAreaTargetAction.apply(lfNetwork, null, acParameters.getNetworkParameters()));
```

Comment:
The behavior should change very soon, after #1156 's merge.

Because of the ContingencyLoadFlowParameters extension, AIC can be performed on some post-contingency cases even if networkParameters.isAreaInterchangeControl() is false. So Area-s will now always be created, and this assertion will fail

Reply:
Thanks I will update on if 1156 is merged before this one.